internal/test/framework/framework.go

```diff
@@ -284,6 +294,67 @@ func (f *TestFramework) GetReleaseArtifactSHA(artifactName string) (string, erro
 	return "", fmt.Errorf("no artifact with name %s", artifactName)
 }
 
+// getLatestCniPluginsVersion returns the latest 0.8.x version of CNI plugins in 'v<major>.<minor>.<patch>' format
+func (f *TestFramework) getLatestCniPluginsVersion() error {
+	client := github.NewClient(nil)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*20)
+	defer cancel()
+	releases, _, err := client.Repositories.ListReleases(ctx, "containernetworking", "plugins",
+		&github.ListOptions{})
+	if err != nil {
+		return err
+	}
+	// Iterating over releases to fetch versions from tag names
+	var cniPluginsVersions = []string{}
+	for _, release := range releases {
+		cniPluginsVersions = append(cniPluginsVersions, release.GetTagName())
+	}
+	// Sorting versions in reverse order so as to get latest version first
+	sort.Sort(sort.Reverse(sort.StringSlice(cniPluginsVersions)))
+	// Iterating over versions to find first 0.8.x version which is not a release candidate
+	for _, cniPluginsVersion := range cniPluginsVersions {
+		if strings.HasPrefix(cniPluginsVersion, "v0.8.") && !strings.Contains(cniPluginsVersion, "rc") {
+			f.LatestCniPluginsVersion = cniPluginsVersion
+			return nil
+		}
+	}
+	return fmt.Errorf("could not fetch latest 0.8.x version of CNI Plugins")
+}
+
+// GetLatestCniPluginsURL returns the URL of the latest 0.8.x version of CNI Plugins
+func (f *TestFramework) GetLatestCniPluginsURL() (string, error) {
+	latestCniPluginsURL := cniPluginsBaseURL + f.LatestCniPluginsVersion + "/cni-plugins-windows-amd64-" +
+		f.LatestCniPluginsVersion + ".tgz"
+	return latestCniPluginsURL, nil
+}
+
+// GetLatestCniPluginsSHA returns the SHA512 of the latest 0.8.x version of CNI Plugins
+func (f *TestFramework) GetLatestCniPluginsSHA() (string, error) {
```

Comment:
Same as above? Can you refactor this if possible?

Reply:
I cannot see any code getting duplicated here